docs/datastream/datastream.md

```diff
@@ -0,0 +1,193 @@
+# DATASTREAM FORMAT
+
+The format of the data in the data stream is defined using protobuf. Latest version can always be found here. This document describes them as they are in the date of creation of the document.
+
+## Entry types
+
+### BookMark
+
+​​​​- Type
+
+​​​​- Value
+
+**Notes**
+There are bookmarks for Batches (type = 1) and L2Blocks (type = 2).
+
+### BatchStart
+
+​​​​- Number
+
+​​​​- Type
+
+​​​​- ForkID
+
+​​​​- ChainID
+
+​​​​- Debug
+
+**Notes**
+
+* ForkID is 0 for genesis batch
+
+* Type can be:
+
+   - Regular: For normal batches
+
+   - Forced: For forced batches
+
+   - Injected: Batch 1 of a network with ForkID >= 7. It is generated by the smart contract during network creation.
+
+
+### L2Block
+
+​​​​- Number
+
+​​​​- BatchNumber
+
+​​​​- Timestamp
+
+​​​​- DeltaTimestamp
+
+​​​​- MinTimestamp
+
+​​​​- L1BlockHash
+
+​​​​- L1InfoTreeIndex
+
+​​​​- Hash (BlockHash)
+
+​​​​- StateRoot
+
+​​​​- GlobalExitRoot
+
+​​​​- Coinbase
+
+​​​​- BlockGasLimit
+
+​​​​- BlockInfoRoot
+
+- StateRoot
+
+​​​​- Debug
+
+**Notes**
+
+* **Hash** - hash calculated by the node and used in the RPC
+
+* **StateRoot** 
+
+  - For ForkIDs < 7: Final state root of the L2Block (see also *IntermediateStateRoot* in *L1Transacton*)
+
+  - For ForkIDs >=7: Final state root of the L2Block
```

Comment:
maybe this can be condensed into a single bullet point with a note about pre-etrog intermediate root